internal/pkg/service/appsproxy/proxy/proxy_test.go

```diff
@@ -90,110 +89,116 @@ func TestAppProxyRouter(t *testing.T) {
 				require.Equal(t, http.StatusNotFound, response.StatusCode)
 				body, err := io.ReadAll(response.Body)
 				require.NoError(t, err)
-				assert.Equal(t, `Application "unknown" not found.`, string(body))
+				assert.Contains(t, string(body), html.EscapeString(`Application "unknown" not found in the stack.`))
 			},
 			expectedNotifications: map[string]int{},
 			expectedWakeUps:       map[string]int{},
 		},
 		{
 			name: "broken-app",
-			run: func(t *testing.T, client *http.Client, m []*mockoidc.MockOIDC, appServer *appServer, service *sandboxesService, dnsServer *dnsmock.Server) {
+			run: func(t *testing.T, client *http.Client, m []*mockoidc.MockOIDC, appServer *testutil.AppServer, service *testutil.DataAppsAPI, dnsServer *dnsmock.Server) {
 				// Request to broken app
 				request, err := http.NewRequestWithContext(context.Background(), http.MethodGet, "https://broken.hub.keboola.local/", nil)
 				require.NoError(t, err)
 				response, err := client.Do(request)
 				require.NoError(t, err)
-				require.Equal(t, http.StatusForbidden, response.StatusCode)
+				require.Equal(t, http.StatusServiceUnavailable, response.StatusCode)
 				body, err := io.ReadAll(response.Body)
```

Comment:
502 bad gateway: problem with the server in the app itself, e.g. it doesn't run
503 service unavailable: problem with the configuration of the app

Reply:
> > I generally tried to prefer 4xx rather than 5xx since you told me that 5xx cause alerts but these cases are typically caused by user configuration of the data app.
> 
> Sorry if I informed you wrongly, ... from the user's point of view, it doesn't matter if the error is in the proxy or in the app, it's 5xx, ... we have to distinguish it in our telemetry and alerts.